lib/temporal/workflow/dispatcher.rb

```diff
@@ -40,19 +35,25 @@ def unregister
       end
 
       WILDCARD = '*'.freeze
-      TARGET_WILDCARD = '*'.freeze
 
       EventStruct = Struct.new(:event_name, :handler)
 
       def initialize
-        @handlers = Hash.new { |hash, key| hash[key] = {} }
+        @event_handlers = Hash.new { |hash, key| hash[key] = {} }
+        @wait_until_handlers = {}
         @next_id = 0
       end
 
       def register_handler(target, event_name, &handler)
         @next_id += 1
-        handlers[target][@next_id] = EventStruct.new(event_name, handler)
-        RegistrationHandle.new(handlers[target], @next_id)
+        event_handlers[target][@next_id] = EventStruct.new(event_name, handler)
+        RegistrationHandle.new(event_handlers[target], @next_id)
+      end
+
+      def register_wait_until_handler(&handler)
```

Comment:
I think we should rename this method into something that isn't coupled to the `wait_until` name since that might change and also there might be other uses of this method in the future. Maybe something like a `register_unscoped_handler` or `register_wildcard_handler`?

Reply:
There are two dimensions to each handler here: the order they run in and which events (ID/target and type) cause them to run. The wildcard aspect applies to both old-style signal handlers and wait_until handlers: they both match to all event targets. When they run is changing with this PR though. Before, the order was prescribed entirely by the order registered. Handlers reacting to a specific event must run first, then wait_until handlers need to run next.

I ended up naming this after the wait_until handlers, because they need this very specific behavior of running after the handlers that target a specific event. I suppose I could introduce a more generic approach here where the caller specifies a "priority" for the handler. When two handlers have the same priority, we would still break ties by using the order they were registered in.

Personally, I find the event-specific and wait_until-specific naming clearer because the ordering behavior is specific to these use cases. Another approach would be to lean into this approach, and move the wait_until handlers out of the dispatcher entirely. The code would be similar to the dispatcher, but much simpler since these do not need to match specific event IDs or types. I think I'd prefer this approach over specifying a priority, because I suspect the code would be easier to follow. I'll add another commit that gives this a try, assuming that it looks good to me otherwise.